yupana/processor/report_slice_processor.py

```diff
@@ -264,7 +264,12 @@ def _remove_empty_ip_addresses(
             host: dict, transformed_obj=copy.deepcopy(TRANSFORMED_DICT)):
         """Remove 'ip_addresses' field."""
         ip_addresses = host.get('ip_addresses')
-        if ip_addresses is None or ip_addresses:
+        if ip_addresses is None:
+            return [host, transformed_obj]
+        if ip_addresses:
```

Comment:
```suggestion
        if ip_addresses is None or len(x) == len(set(x)): 
            return [host, transformed_obj]
        else:
```
Let us first check if list in unique, this way transformed_obj is only appended if ip_address list is really modified.